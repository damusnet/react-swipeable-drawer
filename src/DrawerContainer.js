import React from "react";
import PropTypes from "prop-types";

import { DrawerOverlay, DrawerContentContainer } from "./Drawer";

const DrawerContainer = ({
  position,
  size,
  zIndex,
  swiping,
  translation,
  toggleDrawer,
  handleTouchStart,
  handleTouchMove,
  handleTouchEnd,
  drawerContent,
}) => {
  const open = translation > 0;

  return (
    <div className="DrawerContainer">
      <DrawerOverlay
        position={position}
        zIndex={zIndex}
        open={open}
        swiping={swiping}
        translation={translation}
        toggleDrawer={toggleDrawer}
        handleTouchStart={handleTouchStart}
        handleTouchMove={handleTouchMove}
        handleTouchEnd={handleTouchEnd}
      />
      <DrawerContentContainer
        position={position}
        size={size}
        zIndex={zIndex + 1}
        swiping={swiping}
        translation={translation}
        toggleDrawer={toggleDrawer}
        handleTouchStart={handleTouchStart}
        handleTouchMove={handleTouchMove}
        handleTouchEnd={handleTouchEnd}
        drawerContent={drawerContent}
      />
    </div>
  );
};

export default DrawerContainer;

DrawerContainer.propTypes = {
  position: PropTypes.oneOf(["left", "right", "top", "bottom"]).isRequired,
  size: PropTypes.number.isRequired,
  zIndex: PropTypes.number.isRequired,
  swiping: PropTypes.bool.isRequired,
  translation: PropTypes.number.isRequired,
  toggleDrawer: PropTypes.func.isRequired,
  handleTouchStart: PropTypes.func.isRequired,
  handleTouchMove: PropTypes.func.isRequired,
  handleTouchEnd: PropTypes.func.isRequired,
  drawerContent: PropTypes.element.isRequired,
};
