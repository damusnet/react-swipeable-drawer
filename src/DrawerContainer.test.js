import React from "react";
import renderer from "react-test-renderer";

import DrawerContainer from "./DrawerContainer";

const noop = () => () => {};

describe("<DrawerContainer />", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <DrawerContainer
        position="left"
        size={80}
        zIndex={1}
        swiping={false}
        translation={0}
        toggleDrawer={noop}
        handleTouchStart={noop}
        handleTouchMove={noop}
        handleTouchEnd={noop}
        drawerContent={<div />}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
