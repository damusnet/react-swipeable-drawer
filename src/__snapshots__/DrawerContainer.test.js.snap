// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerContainer /> renders correctly 1`] = `
<div
  className="DrawerContainer"
>
  <div
    className="DrawerOverlay"
    onClick={null}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0)",
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "top": 0,
        "transition": "background-color .2s ease-in-out, width 0s .2s, height 0s 0s",
        "width": "20px",
        "zIndex": 1,
      }
    }
  />
  <div
    className="DrawerContentContainer"
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": "-80%",
        "position": "fixed",
        "top": 0,
        "transform": "translateX(0%)",
        "transition": "transform .2s ease-in-out",
        "width": "80%",
        "zIndex": 2,
      }
    }
  >
    <div />
  </div>
</div>
`;
